tensorflow/python/ops/array_ops.py

```diff
@@ -99,6 +99,21 @@
 listdiff = gen_array_ops.list_diff
 
 
+def rank(input, name=None):
+  """Override for gen_array_ops.rank(), making it work for SparseTensor.
+
+  Args:
+    input: A `Tensor` or `SparseTensor` object.
+    name: A name for the operation (optional).
+
+  Returns:
+    A `Tensor` of type `int32`.
+  """
+  if isinstance(input, ops.SparseTensor):
+    return size(input.shape)
```

Comment:
I think `size()` here refers to `gen_array_ops.size()`?  If so, please qualify it.